### DISCUSSION TITLE
Sunbeam Dual Mattress Detected As Air Purifier

### DISCUSSION THREAD

**Discussion by monsieurlatte** [make-all/tuya-local#2248] [Q&A] [answered]
So I stupidly deleted my old Tuya setup that I had painstakingly added in the DPS and all that and I really didnt want to do that again, I found this repo and it shows the mattress is in there but when I add it in at the end it says pick the device that best matches your device and the only thing in there is Air Purifier (which doesnt work).  Any help would be greatly appreciated.

**Comment by monsieurlatte** [comment]
I'll add if this can be fixed and works in the future I'll buy you $50 worth of coffee as that's how much this is important to my wife and how little time I have to fix this f'n thing lol.

**Reply by make-all** [reply]
What is the message in the Home Assistant log that comes out when this happens?

**Reply by monsieurlatte** [reply]
Assume this is what you're looking for?  And thanks for the reply!

Device matches None with quality of 0%. DPS: {"updated_at": 1724798109.689457, "14": true, "15": false, "20": "level_1", "24": false, "26": "Stay_On", "28": 79080, "105": "ZONE_A"}

**Reply by monsieurlatte** [reply]
Here's a link to the bug I filed just in case there's more info in there that helps.

https://github.com/make-all/tuya-local/issues/2249

**Reply by make-all** [reply]
Yes, that's it. It looks like adding `optional: true` to dps id 8 and 25 will fix the issue.  Possibly you would be able to get it to connect by setting separate temperatures for both sides first, as the missing dps seem to be basic setting for side B,

**Reply by monsieurlatte** [reply]
Got it, I will try setting side B to a temp and then adding it again and see if it finds it.  Anything beyond that that I can do to fix this?

**Reply by monsieurlatte** [reply]
well poop, that didnt work (setting both sides to temps).

**Reply by make-all** [reply, marked as answer]
Beyond that, you can replace the `sunbeam_dual_bedwarmer.yaml` file in `config/custom_components/tuya_local/devices/` with the latest version here: https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/sunbeam_dual_bedwarmer.yaml

**Reply by monsieurlatte** [reply]
That worked, coffee incoming mate!

**Reply by monsieurlatte** [reply]
Just want to note that the Side A Auto off was missing the Stay_On part, but since I saw it on the Side B I just copied it over and replaced the yaml again and it's all working perfectly, just letting you know in case that file gets purged into your repo!

**Reply by make-all** [reply]
Thanks. These timers were also missed in the bulk update for translations, so the half hour increments are not matching the translated strings.
